fdbserver/ClusterController.actor.cpp

```diff
@@ -2458,12 +2469,53 @@ ACTOR Future<Void> monitorRatekeeper(ClusterControllerData *self) {
 			.detail("RKID", self->db.serverInfo->get().ratekeeper.get().id());
 			self->db.clearInterf(ProcessClass::RateKeeperClass);
 		} else {
+			self->recruitingRatekeeper = true;
 			RatekeeperInterface rkInterf = wait( startRatekeeper(self) );
+			self->recruitingRatekeeper = false;
 			self->db.setRatekeeper(rkInterf);
 		}
 	}
 }
 
+ACTOR Future<Void> monitorBetterDDOrRK(ClusterControllerData* self) {
+	state std::map<Optional<Standalone<StringRef>>, WorkerInfo>::iterator masterWorker;
+
+	loop {
+		masterWorker = self->id_worker.find(self->db.serverInfo->get().master.locality.processId());
+		if (masterWorker == self->id_worker.end()) {
+			wait(self->db.serverInfo->onChange());
+			continue;
+		}
+
+		const ServerDBInfo& db = self->db.serverInfo->get();
+		auto masterFitness = masterWorker->second.details.processClass.machineClassFitness(ProcessClass::Master);
+
+		state Future<Void> rkReply = Void();
```

Comment:
check if rk or dd is on the same process as proxies or resolvers, and if so move it to the master